tests/unit_tests/test_import_gaurds.py

```diff
@@ -0,0 +1,98 @@
+# Standard Python modules
+import sys
+import unittest
+from unittest.mock import patch
+import warnings
+
+
+def ignore_warnings(test_func):
+    def do_test(self, *args, **kwargs):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            test_func(self, *args, **kwargs)
+
+    return do_test
+
+
+class TestImportGaurds(unittest.TestCase):
+    N_PROCS = 1
+
+    @ignore_warnings
```

Comment:
Why is this necessary?

Reply:
I think this is because we are forcing the error and do not want testflo to quit - we silently ignore all the exceptions and just check them against the expected exception message

Reply:
Are you referring to the `N_PROCS` or the `@ignore_warnings`?  The `N_PROCS` is because testflo was failing for these tests on some procs in parallel becuase of the way I temporarily modify the system packages to force the exceptions.  The `@ignore_warnings` is because numpy is imported every time we import something for these tests and it has internal warnings about efficiency that kept being raised, so I silenced them

Reply:
I was referring to the `@ignore_warnings`. I see the numpy import warning. The tests still pass without this, but I think it's fine to silence to warnings.